tests/gtest/avifreadimagetest.cc

```diff
@@ -16,43 +16,64 @@ const char* data_path = nullptr;
            {AVIF_CHROMA_DOWNSAMPLING_AUTOMATIC,
             AVIF_CHROMA_DOWNSAMPLING_FASTEST,
             AVIF_CHROMA_DOWNSAMPLING_BEST_QUALITY,
             AVIF_CHROMA_DOWNSAMPLING_AVERAGE}) {
+        SCOPED_TRACE("chroma_downsampling: " +
+                     std::to_string(chroma_downsampling));
         const testutil::AvifImagePtr image1 = testutil::ReadImage(
             data_path, file_name1, requested_format, requested_depth,
-            chroma_downsampling, kIgnoreMetadata, kIgnoreMetadata,
-            kIgnoreMetadata);
+            chroma_downsampling, ignore_metadata, ignore_metadata,
+            ignore_metadata);
+        ASSERT_TRUE(image1);
         const testutil::AvifImagePtr image2 = testutil::ReadImage(
             data_path, file_name2, requested_format, requested_depth,
-            chroma_downsampling, kIgnoreMetadata, kIgnoreMetadata,
-            kIgnoreMetadata);
-        if (!image1 || !image2 || !testutil::AreImagesEqual(*image1, *image2)) {
-          return false;
+            chroma_downsampling, ignore_metadata, ignore_metadata,
+            ignore_metadata);
+        ASSERT_TRUE(image2);
+        const double psnr = testutil::GetPsnr(*image1, *image2,
+                                              /*ignore_alpha=*/false);
+        const std::string dir = testing::TempDir();
+        const std::string dst1 = dir + "/" + file_name1;
+        const std::string dst2 = dir + "/" + file_name2;
+        if (psnr < min_psnr) {
+          testutil::WriteImage(image1.get(), dst1.c_str());
+          testutil::WriteImage(image2.get(), dst2.c_str());
         }
+        ASSERT_GE(psnr, min_psnr) << "Actual images saved to " << dst1
+                                  << " and " << dst2 << " psnr: " << psnr;
       }
     }
   }
-  return true;
 }
 
 TEST(JpegTest, ReadAllSubsamplingsAndAllBitDepths) {
-  EXPECT_TRUE(AreSamplesEqualForAllReadSettings(
-      "paris_exif_xmp_icc.jpg", "paris_exif_orientation_5.jpg"));
+  AreSamplesEqualForAllReadSettings("paris_exif_xmp_icc.jpg",
+                                    "paris_exif_orientation_5.jpg",
+                                    /*ignoreMetadata=*/true, /*min_psnr=*/99.);
 }
 
 TEST(PngTest, ReadAllSubsamplingsAndAllBitDepths) {
-  EXPECT_TRUE(AreSamplesEqualForAllReadSettings(
-      "paris_icc_exif_xmp.png", "paris_icc_exif_xmp_at_end.png"));
+  AreSamplesEqualForAllReadSettings("paris_icc_exif_xmp.png",
+                                    "paris_icc_exif_xmp_at_end.png",
+                                    /*ignoreMetadata=*/true, /*min_psnr=*/99.);
+  AreSamplesEqualForAllReadSettings("gray_gama_chrm.png",
+                                    "gray_gama_applied.png",
+                                    /*ignoreMetadata=*/false, /*min_psnr=*/99.);
+  AreSamplesEqualForAllReadSettings("gray16_gama_chrm.png",
+                                    "gray16_gama_applied.png",
+                                    /*ignoreMetadata=*/false, /*min_psnr=*/65.);
```

Comment:
Please add a comment on why these are expected to differ.

Also consider adding a `max_psnr` to make sure they differ.